language/bytecode-verifier/src/dependencies.rs

```diff
@@ -354,62 +354,42 @@ impl CyclicModuleDependencyChecker {
         module: &CompiledModule,
         immediate_module_dependencies: F,
     ) -> PartialVMResult<()> {
-        // A variant of the three-color DFS, visit each node at max once.
-        // This function returns true if a back-edge is found during the DFS.
-        fn dfs_visit<F: Fn(&ModuleId) -> Vec<ModuleId>>(
-            cursor_module: ModuleId,
-            modules_discovered: &mut BTreeSet<ModuleId>,
-            modules_finished: &mut BTreeSet<ModuleId>,
+        fn check_existence_in_dependency_recursive<F: Fn(&ModuleId) -> Vec<ModuleId>>(
+            target_module_id: &ModuleId,
+            cursor_module_id: &ModuleId,
             immediate_module_dependencies: &F,
-        ) -> bool {
-            modules_discovered.insert(cursor_module.clone());
-            for next_module in immediate_module_dependencies(&cursor_module) {
-                if modules_finished.contains(&next_module) {
-                    continue;
-                }
-                if modules_discovered.contains(&next_module)
-                    || dfs_visit(
-                        next_module,
-                        modules_discovered,
-                        modules_finished,
+            visited_modules: &mut BTreeSet<ModuleId>,
+        ) -> PartialVMResult<bool> {
```

Comment:
Remove result?

Reply:
My bad... Right fixing it.

Reply:
Done, but it can be further simplified to
```
cursor_module_id == target_module_id
    || (visited_modules.insert(cursor_module_id.clone())
        && immediate_module_dependencies(cursor_module_id)
            .into_iter()
            .any(|next| {
                check_existence_in_dependency_recursive(
                    target_module_id,
                    &next,
                    immediate_module_dependencies,
                    visited_modules,
                )
            }))
```

But not this might break readability...

Reply:
oof, yeah I think it is alright as is